source/use-cases/local-master-key.txt

```diff
@@ -1,23 +1,48 @@
 
-This strategy bolsters the security, minimizes encryption/decryption overhead, and allows the data encryption keys to be stored in the same replica set as the encrypted data. If a basic encryption strategy was used instead of envelope encryption, the computational and network load of encryption and decryption would need to be handled by the system that manages your keys. For more information on envelope encryption and key management concepts, read `this AWS article <https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#enveloping>`_.
+The master key, used by the MongoDB driver to create and encrypt data keys, should be stored remotely in a Key Management Service (KMS). The data encryption keys, generated and used by the MongoDB driver to encrypt and decrypt document fields, are stored in a key vault collection in the same database as the encrypted data.
 
-The master key is stored separately in a Key Management System (KMS) in order to prevent the data encryption keys from being used directly to access the sensitive data.
+In the following example, we forgo using a KMS for the master key and instead generate a local one for testing purposes. This local master key is a 96-byte, randomly generated string saved to a file.
 
-The data encryption keys can safely be stored in a MongoDB collection alongside the encrypted sensitive data fields since they have been encrypted by the master key.
-
-In Client-Side Field Level Encryption, a client that can access both master key and data encryption keys is necessary to perform encryption/decryption.
-
-In the steps below, we set up a local master key saved to a file and load it into the MongoDB client settings in your development environment.
+If you have access to AWS KMS and wish to skip this testing step, refer to :ref:`Convert to a Remote Master Key` instead.
 
 .. admonition:: Local Master Keys Are Not Secure
    :class: important
-   
+
    To ensure that the master key cannot be compromised, do not use a local master key in a production environment. Instead, use a secure `Key Management System <https://en.wikipedia.org/wiki/Key_management#Key_management_system>`_ such as `AWS KMS <https://aws.amazon.com/kms/>`_.
-   
-   We will demonstrate how to transition from a local key to AWS KMS in a later step of this guide.
 
-.. include:: /includes/steps/create-local-master-key.rst
+   We demonstrate how to transition from a locally-hosted master key to a remote AWS KMS master key in a later step of this guide.
+
+.. tabs::
+
+  tabs:
+
+    - id: java-master-key-generator
+      name: "Java"
+      content: |
+
+        The following script generates a 96-byte local master key and saves it to a file called ``master-key.txt`` in the directory from which the script is executed.
```

Comment:
Currently we provide this same info twice in quick succession - let's pick one spot

Reply:
Removed the detail from the prior mention.